### DISCUSSION TITLE
Mapping to model with parent reference when DTO does not have that reference as a property

### DISCUSSION THREAD

**Discussion by andyfurniss4** [riok/mapperly#1066] [Q&A]
I've been trying to figure out whether Mapperly can solve my Domain/DTO mapping issues but I'm struggling to answer that question right now. I've never used Mapperly before but it looks appealing. Basically, I have a domain model, with child models where the child models have a reference to the parent. However, the DTO does not have these references as it's not necessary to persist them since it's all stored as a single nested document in a document store.

My models look something like this:

```csharp
public class ParentDomain
{
    public Guid Id { get; }
    public Guid BlueprintId { get; }
    public IEnumerable<ParentDomain> Children { get; }

    public ParentDomain(Blueprint blueprintModel)
    {
        Id = Guid.NewGuid();
        BlueprintId = blueprintModel.Id;
        Children = blueprintModel.Things.Select(thing => new ChildDomain(this, thing.Name));
    }
}

public class ChildDomain
{
    public string Name { get; }
    public ParentDomain Parent { get; }

    public ChildDomain(ParentDomain parent, string name)
    {
        Parent = parent;
        Name = name;
    }

    public void DoSomethingWithParent()
    {
        // ...
    }
}

public record ParentDto(Guid Id, Guid BlueprintId, IEnumerable<ChildDto> Children);
public class ChildDto(string Name);
```

I want to be able to map from `ParentDto` to `ParentDomain` when retrieving data from persistence, but I'm not sure how to resolve the Child -> Parent reference, since the DTO does not have a parent reference. Is this possible with Mapperly?

This discussion seems similar but I was struggling to find myself an answer there: https://github.com/riok/mapperly/discussions/849

**Comment by latonz** [comment]
This looks like a pretty advanced mapping that is not supported by Mapperly. You may want to implement it yourself.
